docs/patterns.md

```diff
@@ -142,3 +142,132 @@ Example access policy:
 * The front desk cannot cut releases nor deploy services
 
 Complex policies like this are currently best implemented in code directly, though there is [ongoing work](https://github.com/michaelansel/hubot-rbac) to build a generalized framework for access management.
+
+## Middleware Examples
+
+Middleware is often used for authentication and authorization.  There are three kinds of middleware:
```

Comment:
That's a really good point. The hard part with middleware, I think, is that you'd need some examples to really explore how they work. I could see having some very superficial examples in scripting.md, and then go into more detail in patterns.md to describe some things you'd commonly want to do (as is here already).